### PR TITLE
Add a fqn field to blogs and users

### LINKED ISSUE #319: Cache FQNs in DB
We are recomputing them so many time when they would just be stored as regular field in the database.

Plume version (and/or commit): 0.2.0

### PR DESCRIPTION
**Pull request by BaptisteGelez**
Fixes #319

### CODE REVIEW

**Comment by fdb-hiroshima** [comment]
I'm wondering. On a brand new, uninitialized instance, does it works? The select would return 0 rows, so does the CASE never match, or does it fail?

**Comment by BaptisteGelez** [comment]
I think it will never run, since there are no blogs or users, so no problem.

File: migrations/sqlite/2019-03-05-082846_add_fqn/up.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -- Your SQL goes here
 ALTER TABLE blogs ADD COLUMN fqn TEXT NOT NULL DEFAULT '';
 UPDATE blogs SET fqn =
-	(CASE WHEN instance_id = 1 THEN
+	(CASE WHEN (SELECT local FROM instances WHERE id = instance_id) THEN
 		actor_id
 	ELSE
 		(actor_id || '@' || (SELECT public_domain FROM instances WHERE id = instance_id LIMIT 1))
```
Comment (fdb-hiroshima) [inline migrations/sqlite/2019-03-05-082846_add_fqn/up.sql]: why LIMIT 1? Isn't id unique?

Comment (BaptisteGelez) [reply migrations/sqlite/2019-03-05-082846_add_fqn/up.sql]: Just to be sure. :stuck_out_tongue:

File: migrations/sqlite/2019-03-05-082846_add_fqn/down.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+-- This file should undo anything in `up.sql`
+ALTER TABLE blogs DROP COLUMN fqn;
+ALTER TABLE users DROP COLUMN fqn;
```
Comment (fdb-hiroshima) [inline migrations/sqlite/2019-03-05-082846_add_fqn/down.sql]: sqlite doesn't allow to drop column, you have to do the annoying copy thing

Comment (BaptisteGelez) [reply migrations/sqlite/2019-03-05-082846_add_fqn/down.sql]: Ah, yes, forgot about that… :expressionless:

File: migrations/sqlite/2019-03-05-082846_add_fqn/down.sql [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+    actor_id VARCHAR NOT NULL,
+    title VARCHAR NOT NULL,
+    summary TEXT NOT NULL DEFAULT '',
+    outbox_url VARCHAR NOT NULL UNIQUE,
+    inbox_url VARCHAR NOT NULL UNIQUE,
+    instance_id INTEGER REFERENCES instances(id) ON DELETE CASCADE NOT NULL,
+    creation_date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    ap_url text not null default '' UNIQUE,
+    private_key TEXT,
+    public_key TEXT NOT NULL DEFAULT '', fqn TEXT NOT NULL DEFAULT '',
```
Comment (fdb-hiroshima) [inline migrations/sqlite/2019-03-05-082846_add_fqn/down.sql]: `blogs_no_fqn`, `fqn TEXT NOT NULL`...

File: migrations/sqlite/2019-03-05-082846_add_fqn/down.sql [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+    hashed_password TEXT,
+    instance_id INTEGER REFERENCES instances(id) ON DELETE CASCADE NOT NULL,
+    creation_date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    ap_url TEXT NOT NULL default '' UNIQUE,
+    private_key TEXT,
+    public_key TEXT NOT NULL DEFAULT '',
+    shared_inbox_url VARCHAR,
+    followers_endpoint VARCHAR NOT NULL DEFAULT '' UNIQUE,
+    avatar_id INTEGER REFERENCES medias(id) ON DELETE SET NULL,
+    last_fetched_date TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP, fqn TEXT NOT NULL DEFAULT '',
```
Comment (fdb-hiroshima) [inline migrations/sqlite/2019-03-05-082846_add_fqn/down.sql]: `users_no_fqn`, `fqn TEXT NOT NULL`...

File: migrations/sqlite/2019-03-05-082846_add_fqn/down.sql (hunk 3 above)
Comment (BaptisteGelez) [reply migrations/sqlite/2019-03-05-082846_add_fqn/down.sql]: Ah, that's why it was telling me I was not supplying enough values when doing `INSERT INTO blogs_no_fqn SELECT all, the, columns, but, not, fqn FROM blogs;`. :grimacing:

File: plume-models/src/schema.rs [hunk 5]
```diff
@@ -47,160 +47,160 @@ table! {
     }
 }
 
+table! {
+    comment_seers (id) {
+        id -> Nullable<Integer>,
```
Comment (fdb-hiroshima) [inline plume-models/src/schema.rs]: this is not supposed to be Nullable, but I don't get where it come from, your migrations aren't touching this table

Comment (BaptisteGelez) [reply plume-models/src/schema.rs]: Diesel is regenerating the schema when running migrations, so it changed when I tested with SQlite.

Comment (fdb-hiroshima) [reply plume-models/src/schema.rs]: For some reason, comment_seers's id is an `INTEGER PRIMARY KEY AUTOINCREMENT`, not an `INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT` on sqlite (but fine on pg I assume), that's why the scheme was invalid

**Pull request merged**